esdb/impl.go

```diff
@@ -420,59 +414,22 @@ func getSupportedMethods(ctx context.Context, conf *Configuration, conn *grpc.Cl
 	return &info, nil
 }
 
-type perCallCredentials interface {
-	setCallCredentials(c *Credentials)
-}
-
-type basicPerCallAuth struct {
-	defaultAuth string
-	callAuth    string
-}
-
-func NewBasicPerCallAuth(username, password string) *basicPerCallAuth {
-	auth := &basicPerCallAuth{
-		defaultAuth: "",
-		callAuth:    "",
-	}
-	auth.setDefaultCredentials(username, password)
-
-	return auth
-}
-
-func (b *basicPerCallAuth) GetRequestMetadata(tx context.Context, in ...string) (map[string]string, error) {
-	md := map[string]string{}
-
-	if b.callAuth != "" {
-		md["Authorization"] = b.callAuth
-	} else if b.defaultAuth != "" {
-		md["Authorization"] = b.defaultAuth
-	}
-
-	return md, nil
-}
-
-func (*basicPerCallAuth) RequireTransportSecurity() bool {
-	return false
+func newBasicAuthPerRPCCredentials(username, password string) *basicAuthPerRPCCredentials {
+	authorization := "Basic " + base64.StdEncoding.EncodeToString([]byte(username+":"+password))
+	headers := map[string]string{"Authorization": authorization}
+	return &basicAuthPerRPCCredentials{headers: headers}
 }
 
-func (b *basicPerCallAuth) setDefaultCredentials(username, password string) {
-	if username != "" {
-		b.defaultAuth = b.getAuth(username, password)
-	}
+type basicAuthPerRPCCredentials struct {
+	headers map[string]string
 }
 
-func (b *basicPerCallAuth) setCallCredentials(c *Credentials) {
-	if c != nil {
-		b.callAuth = b.getAuth(c.Login, c.Password)
-	} else {
-		b.callAuth = ""
-	}
+func (b *basicAuthPerRPCCredentials) GetRequestMetadata(_ context.Context, _ ...string) (map[string]string, error) {
+	return b.headers, nil
 }
 
-func (*basicPerCallAuth) getAuth(username, password string) string {
-	auth := username + ":" + password
-	enc := base64.StdEncoding.EncodeToString([]byte(auth))
-	return "Basic " + enc
+func (*basicAuthPerRPCCredentials) RequireTransportSecurity() bool {
+	return false // Shouldn't this be true? Don't want to send password over insecure connection?
```

Comment:
I think you are right, we could detect upfront if we are in insecure mode and return that value instead.

Reply:
I changed this to return true.